flink-batch-connectors/flink-jdbc/src/main/java/org/apache/flink/api/java/io/jdbc/JDBCInputFormat.java

```diff
@@ -232,7 +255,7 @@ private void addValue(OUT reuse) throws IOException, SQLException {
 						reuse.setField(resultSet.getInt(pos + 1), pos);
 						break;
 					case java.sql.Types.FLOAT:
-						reuse.setField(resultSet.getDouble(pos + 1), pos);
+						reuse.setField(resultSet.getFloat(pos + 1), pos);
```

Comment:
I'm aware of the confusion. However that is something users will have to live with; we can't just change the data so that a user doesn't run into an exception he _should_ run into.

All we can do is document the mapping and hope that people will read it.